src/typescript/frontend/src/components/pages/pools/ClientPoolsPage.tsx

```diff
@@ -16,9 +18,23 @@ import {
   StyledInner,
   StyledSubHeader,
 } from "components/pages/pools/styled";
-import { isDisallowedEventKey } from "utils";
+import { isDisallowedEventKey, parseJSON } from "utils";
+import type { SortByPageQueryParams } from "lib/queries/sorting/types";
+
+const fetcher = (...args: Parameters<typeof fetch>) =>
+  fetch(...args)
+    .then((res) => res.text())
+    .then((txt) => parseJSON(txt));
 
 export const ClientPoolsPage = () => {
+  const [sortBy, setSortBy] = useState<SortByPageQueryParams>("all_time_vol");
+  const [orderBy, setOrderBy] = useState<"desc" | "asc">("desc");
+  const [selectedIndex, setSelectedIndex] = useState<number>();
+  const { data, error, isLoading } = useSWR(
+    `/pools/api?sortby=${sortBy}&orderby=${orderBy}`,
+    fetcher
```

Comment:
Yes, exactly! Basically what you already have with `swr` can go into a server action and be passed down from server component to the client as prop data, and then you can use `tanstack` to manage any custom querise for the user.

You could also just put the `unstable_cache` (in the form of `cached` in our repo, sorry for any alias confusion...) call in a server action and then use that in the client, too.

The primary motivator for this is lots of users are going to sort and get the same exact data when querying with the pools filters- so we can cache it for some X amount of seconds per user instead of having each of them call the postgrest API independently.

Basically, at least from my understanding, with `unstable_cache` the nextjs server acts as middleware and stores the request in the Data Cache rather than it being stored in a client-side cache with tanstack/swr

https://nextjs.org/docs/app/building-your-application/caching

Reply:
Theoretically you should be able to use the server action in a client component actually- but I've run into a lot of issues trying to get it to work. I think it's because we're using React 18 instead of the canary releases (react 19 and nextjs 15)